core/primitives-core/src/runtime/fees.rs

```diff
@@ -203,28 +204,35 @@ impl StorageUsageConfig {
 
 /// Helper functions for computing Transfer fees.
 /// In case of implicit account creation they always include extra fees for the CreateAccount and
-/// AddFullAccessKey actions that are implicit.
+/// AddFullAccessKey (for NEAR-implicit account only) actions that are implicit.
 /// We can assume that no overflow will happen here.
-pub fn transfer_exec_fee(cfg: &RuntimeFeesConfig, is_receiver_implicit: bool) -> Gas {
+pub fn transfer_exec_fee(
+    cfg: &RuntimeFeesConfig,
+    is_receiver_implicit: bool,
+    receiver_account_type: AccountType,
+) -> Gas {
+    let mut result = cfg.fee(ActionCosts::transfer).exec_fee();
```

Comment:
How would you feel about a match statement and some extra comments to clarify what fees need to be applied for what account type? I think, it would be slightly easier to read even at the cost of duplicating a line.